recipes/tree-gen/all/test_package/conanfile.py

```diff
@@ -0,0 +1,31 @@
+import os
+
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.cmake import CMake, cmake_layout
+
+
+class TreeGenTestConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "CMakeDeps", "CMakeToolchain", "VirtualBuildEnv", "VirtualRunEnv"
+    apply_env = False
+    test_type = "explicit"
+
+    def build_requirements(self):
+        self.tool_requires(self.tested_reference_str)
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
```

Comment:
This will need a recompilation for the tests for the same recipe revision to work, we'll handle that

Reply:
Is there something to do here then on my side, @RubenRBS ?